### PR TITLE
Add DisplaySetting component.

### PR DESCRIPTION
**Pull request by jqlee85**
## Summary

Addresses issue [#1296](https://github.com/google/site-kit-wp/issues/1296)

## Relevant technical choices

Adds a DisplaySetting component which takes `value` as a prop, returning the value if it is valid, or `"\u00A0"` if it is falsy. The component is used in `TagmanagerSetup` to display setting values.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/components/display-setting.test.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+/**
+ * Internal dependencies
+ */
+import DisplaySetting from './display-setting';
+import { render } from '../../../tests/js/test-utils';
+
+describe( 'DisplaySetting', () => {
+	it( 'returns non-empty string unchanged', () => {
+		const { container } = render( <DisplaySetting value="Test Value" /> );
+		expect( container.textContent ).toBe( 'Test Value' );
```
Comment (aaemnnosttv) [inline assets/js/components/display-setting.test.js:28]: React Testing Library uses [`jest-dom`](https://github.com/testing-library/jest-dom) so we can actually use a more expressive matcher for this like so:
```suggestion
		expect( container ).toHaveTextContent( 'Test Value' );
```

Would you please update these tests to use this instead?

**Comment by aaemnnosttv** [comment]
@felixarntz this is beyond the scope of the original issue, but do you think we should extend this to the other modules now or do it as part of the rewrite for each as we transition to the datastore? It probably would be relatively low effort to do. What do you think?

**Comment by felixarntz** [comment]
@aaemnnosttv Let's keep it scoped to GTM for now, and let's open an issue to expand to other modules.

File: assets/js/components/display-setting.test.js (hunk 1 above)
Comment (jqlee85) [reply assets/js/components/display-setting.test.js:28]: Yeah, that's the method I tried originally, but `toHaveTextContent()` either normalizes white space or replaces `&nbsp;`s (see reference below) which prevents us from being able to test for that `&nbsp;`. Testing with `textContent` was a workaround. We could use `toHaveTextContent()` for the first case with the non-empty string though. Let me know if I should change that or keep them consistent within this file. (Or if there's a preferred workaround).

[https://github.com/testing-library/jest-dom/blob/d6655dbd0736f9ab8391b59570a0ecd631c93916/src/to-have-text-content.js#L11-L13](https://github.com/testing-library/jest-dom/blob/d6655dbd0736f9ab8391b59570a0ecd631c93916/src/to-have-text-content.js#L11-L13)

Comment (aaemnnosttv) [reply assets/js/components/display-setting.test.js:28]: That makes sense @jqlee85. I see there is an option to turn that off with `{ normalizeWhitespace: false }` as the second parameter to the matcher, but that's a bit overly verbose for each I think.

On its face, I like your approach better as that's still quite clear and is less verbose but the text content-specific matcher gives us more specific feedback in the event of a failure.

Let's go with the `toHaveTextContent()` matcher for the better feedback (unless there's another reason why we can't), but also for consistency with how we make assertions on text content. We can DRY up the usage with a variable for `options` above that they can all share. Maybe this variable could even be called `preserveWhitespace` to be clear about what it's doing?

If for some reason we still can't use this matcher, then let's just document the exception with a comment that explains why 😄

Comment (jqlee85) [reply assets/js/components/display-setting.test.js:28]: @aaemnnosttv the `{normalizeWhitespace: false}` option doesn't resolve the issue, unfortunately. It just causes toHaveTextContent() to explicitly replace **nbsp**s with spaces:

`htmlElement.textContent.replace(/\u00a0/g, ' ') // Replace &nbsp; with normal spaces`.

It's as if toHaveTextContent() was specifically designed to break this test :laughing:.

I've updated the test to use toHaveTextContent for the first case, and added comments to the rest to document why we're using toBe instead.

**Pull request merged**